### ISSUE TITLE
[Android] Failed download video from some Url

### ISSUE THREAD

**Issue by bladeofgod** [781flyingdutchman/background_downloader#406]
**Describe the bug**
Hi, i maybe got a new bug on android, i download video always failed from this url on Android. Ios is fine.

**To Reproduce**
use the url below, and download on android.

```
https://wxapp.tc.qq.com/251/20302/stodownload?encfilekey=Cvvj5Ix3eewK0tHtibORqcsqchXNh0Gf3sJcaYqC2rQA5oZNqQcGRjoPUUBJpBXXsZEffOoc3c2nH4icTZ2bwWd1EHgbOPI73sHyJrEWkd5Nh1ACvMaMVkJtKpnibJuyREF&token=Cvvj5Ix3eexZiajDdmtxmMNuodNwMD6lCuBuX8tKRkn7CqjHzGEGl1PAf1lQkykcTDDE70kE8DJicaCA0DmRgPUicTz2TRpWl902IsJT9oQsOjNKrRgKiaJWSzPuyt0jp4do&idx=1&dotrans=0&hy=SH&m=&uzid=2
```

**Expected behavior**
success download on android.

**Logs**
If possible, include logs that capture the issue:
```
W/TaskWorker(29678): Error for taskId 804481779: Unexpected status line: ������ ftypisom������isomiso2avc1mp41��qQmoov������lmvhd�������������������������������VR����������������������������������������������������������������������������������������@������������������������������������������������������������������trak������\tkhd��������������������������������������VR��������������������������������������������������������������������������������������������@�����������������������$edts������elst����������������VR�����������������_mdia������ mdhd����������������������������<�������U�����������1hdlr����������������vide������������������������Core Media Video�������minf������vmhd����������������������������$dinf������dref��������������������url ������������stbl�������stsd���������������������avc1���������������������������������������������������H������H����������������������������������������������������������������������������������������8avcCd������gd����`-������������ ������3@��h�x�������������btrt���[��5������������stts������������������
W/TaskWorker(29678): java.net.ProtocolException: Unexpected status line: ������ ftypisom������isomiso2avc1mp41��qQmoov������lmvhd�������������������������������VR����������������������������������������������������������������������������������������@������������������������������������������������������������������trak������\tkhd��������������������������������������VR��������������������������������������������������������������������������������������������@�����������������������$edts������elst����������������VR�����������������_mdia������ mdhd����������������������������<�������U�����������1hdlr����������������vide������������������������Core Media Video�������minf������vmhd����������������������������$dinf������dref��������������������url ������������stbl�������stsd���������������������avc1���������������������������������������������������H������H����������������������������������������������������������������������������������������8avcCd������gd����`-������������ ������3@��h�x�������������btrt���[��5������������stts������������������
W/TaskWorker(29678): 	at com.android.okhttp.internal.http.StatusLine.parse(StatusLine.java:56)
W/TaskWorker(29678): 	at com.android.okhttp.internal.http.Http1xStream.readResponse(Http1xStream.java:188)
W/TaskWorker(29678): 	at com.android.okhttp.internal.http.Http1xStream.readResponseHeaders(Http1xStream.java:129)
W/TaskWorker(29678): 	at com.android.okhttp.internal.http.HttpEngine.readNetworkResponse(HttpEngine.java:774)
W/TaskWorker(29678): 	at com.android.okhttp.internal.http.HttpEngine.readResponse(HttpEngine.java:646)
W/TaskWorker(29678): 	at com.android.okhttp.internal.huc.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:475)
W/TaskWorker(29678): 	at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getResponse(HttpURLConnectionImpl.java:411)
W/TaskWorker(29678): 	at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getResponseCode(HttpURLConnectionImpl.java:542)
W/TaskWorker(29678): 	at com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.getResponseCode(DelegatingHttpsURLConnection.java:106)
W/TaskWorker(29678): 	at com.android.okhttp.internal.huc.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:30)
W/TaskWorker(29678): 	at com.bbflight.background_downloader.DownloadTaskWorker.process(DownloadTaskWorker.kt:67)
W/TaskWorker(29678): 	at com.bbflight.background_downloader.TaskWorker.connectAndProcess$suspendImpl(TaskWorker.kt:519)
W/TaskWorker(29678): 	at com.bbflight.background_downloader.TaskWorker.connectAndProcess(Unknown Source:0)
W/TaskWorker(29678): 	at com.bbflight.background_downloader.DownloadTaskWorker.connectAndProcess(DownloadTaskWorker.kt:49)
W/TaskWorker(29678): 	at com.bbflight.background_downloader.TaskWorker.doTask(TaskWorker.kt:494)
W/TaskWorker(29678): 	at com.bbflight.background_downloader.TaskWorker.access$doTask(TaskWorker.kt:43)
W/TaskWorker(29678): 	at com.bbflight.background_downloader.TaskWorker$doWork$2.invokeSuspend(TaskWorker.kt:418)
W/TaskWorker(29678): 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
W/TaskWorker(29678): 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
W/TaskWorker(29678): 	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:115)
W/TaskWorker(29678): 	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:100)
W/TaskWorker(29678): 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
W/TaskWorker(29678): 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
W/TaskWorker(29678): 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
W/TaskWorker(29678): 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)
D/TrafficStats(29678): tagSocket(157) with statsTag=0xffffffff, statsUid=-1
I/System.out(29678): [OkHttp] sendRequest>>
I/System.out(29678): [OkHttp] sendRequest<<
```

**Comment by bladeofgod** [comment]
I reproduce it and see something in log, check out : 

1. first start 3 task : 853124564, 1652368664, 28401318.

watch the id '853124564'.

```
I/BackgroundDownloader(26605): Enqueuing task with id 853124564
I/BackgroundDownloader(26605): Enqueuing task with id 1652368664
I/BackgroundDownloader(26605): Enqueuing task with id 28401318
I/TaskWorker(26605): Starting task with taskId 853124564
I/TaskWorker(26605): Starting task with taskId 1652368664
I/TaskWorker(26605): Starting task with taskId 28401318
I/System.out(26605): [okhttp]:check permission begin!
I/System.out(26605): [okhttp]:not MMS!
I/System.out(26605): [okhttp]:not Email!
I/System.out(26605): [OkHttp] sendRequest>>
I/System.out(26605): [OkHttp] sendRequest<<
W/TaskWorker(26605): Error for taskId 853124564: Unexpected status line: ������ ftypisom������isomiso2avc1mp41����Q�moov������lmvhd���������������������������������^�����������������������������������������������������������������������������������������@����������������������������������������������������������������>Ntrak������\tkhd����������������������������������������^���������������������������������������������������������������������������������������������@�����������������������$edts������elst������������������^�����������������=�mdia������ mdhd����������������������������<�������U�����������1hdlr����������������vide������������������������Core Media Video������=mminf������vmhd����������������������������$dinf������dref��������������������url ����������=-stbl�������stsd���������������������avc1���������������������������������������������������H������H����������������������������������������������������������������������������������������7avcCd������gd����`-������������ ������3@��h�{������������btrt��i����NX���M������stts�������������������������������(stss��������������������������=�����������I���������3���� ctts�����������������������������������������������������������������������
W/TaskWorker(26605): java.net.ProtocolException: Unexpected status line: ������ ftypisom������isomiso2avc1mp41����Q�moov������lmvhd���������������������������������^�����������������������������������������������������������������������������������������@����������������������������������������������������������������>Ntrak������\tkhd����������������������������������������^���������������������������������������������������������������������������������������������@�����������������������$edts������elst������������������^�����������������=�mdia������ mdhd����������������������������<�������U�����������1hdlr����������������vide������������������������Core Media Video������=mminf������vmhd����������������������������$dinf������dref��������������������url ����������=-stbl�������stsd���������������������avc1���������������������������������������������������H������H����������������������������������������������������������������������������������������7avcCd������gd����`-������������ ������3@��h�{������������btrt��i����NX���M������stts�������������������������������(stss��������������������������=�����������I���������3���� ctts�����������������������������������������������������������������������
W/TaskWorker(26605): 	at com.android.okhttp.internal.http.StatusLine.parse(StatusLine.java:56)
W/TaskWorker(26605): 	at com.android.okhttp.internal.http.Http1xStream.readResponse(Http1xStream.java:188)
W/TaskWorker(26605): 	at com.android.okhttp.internal.http.Http1xStream.readResponseHeaders(Http1xStream.java:129)
W/TaskWorker(26605): 	at com.android.okhttp.internal.http.HttpEngine.readNetworkResponse(HttpEngine.java:774)
W/TaskWorker(26605): 	at com.android.okhttp.internal.http.HttpEngine.readResponse(HttpEngine.java:646)
W/TaskWorker(26605): 	at com.android.okhttp.internal.huc.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:475)
W/TaskWorker(26605): 	at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getResponse(HttpURLConnectionImpl.java:411)
W/TaskWorker(26605): 	at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getResponseCode(HttpURLConnectionImpl.java:542)
W/TaskWorker(26605): 	at com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.getResponseCode(DelegatingHttpsURLConnection.java:106)
W/TaskWorker(26605): 	at com.android.okhttp.internal.huc.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:30)
W/TaskWorker(26605): 	at com.bbflight.background_downloader.DownloadTaskWorker.process(DownloadTaskWorker.kt:67)
W/TaskWorker(26605): 	at com.bbflight.background_downloader.TaskWorker.connectAndProcess$suspendImpl(TaskWorker.kt:519)
W/TaskWorker(26605): 	at com.bbflight.background_downloader.TaskWorker.connectAndProcess(Unknown Source:0)
W/TaskWorker(26605): 	at com.bbflight.background_downloader.DownloadTaskWorker.connectAndProcess(DownloadTaskWorker.kt:49)
W/TaskWorker(26605): 	at com.bbflight.background_downloader.TaskWorker.doTask(TaskWorker.kt:494)
W/TaskWorker(26605): 	at com.bbflight.background_downloader.TaskWorker.access$doTask(TaskWorker.kt:43)
W/TaskWorker(26605): 	at com.bbflight.background_downloader.TaskWorker$doWork$2.invokeSuspend(TaskWorker.kt:418)
W/TaskWorker(26605): 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
W/TaskWorker(26605): 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
W/TaskWorker(26605): 	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:115)
W/TaskWorker(26605): 	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:100)
W/TaskWorker(26605): 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
W/TaskWorker(26605): 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
W/TaskWorker(26605): 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
W/TaskWorker(26605): 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)
I/System.out(26605): [okhttp]:check permission begin!
I/System.out(26605): [okhttp]:not MMS!
I/System.out(26605): [okhttp]:not Email!
D/TrafficStats(26605): tagSocket(152) with statsTag=0xffffffff, statsUid=-1
I/System.out(26605): [okhttp]:check permission begin!
I/System.out(26605): [okhttp]:not MMS!
I/System.out(26605): [okhttp]:not Email!
D/TrafficStats(26605): tagSocket(181) with statsTag=0xffffffff, statsUid=-1
I/WM-WorkerWrapper(26605): Work [ id=aaaeaad4-db3f-4f26-8183-2d9c9cb615e8, tags={ com.bbflight.background_downloader.DownloadTaskWorker, BackgroundDownloader, taskId=853124564, group=chunk } ] was cancelled
I/WM-WorkerWrapper(26605): java.util.concurrent.CancellationException: Task was cancelled.
I/WM-WorkerWrapper(26605): 	at androidx.work.impl.utils.futures.AbstractFuture.cancellationExceptionWithCause(AbstractFuture.java:1183)
I/WM-WorkerWrapper(26605): 	at androidx.work.impl.utils.futures.AbstractFuture.getDoneValue(AbstractFuture.java:513)
I/WM-WorkerWrapper(26605): 	at androidx.work.impl.utils.futures.AbstractFuture.get(AbstractFuture.java:474)
I/WM-WorkerWrapper(26605): 	at androidx.work.impl.WorkerWrapper$2.run(WorkerWrapper.java:316)
I/WM-WorkerWrapper(26605): 	at androidx.work.impl.utils.SerialExecutorImpl$Task.run(SerialExecutorImpl.java:96)
I/WM-WorkerWrapper(26605): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
I/WM-WorkerWrapper(26605): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
I/WM-WorkerWrapper(26605): 	at java.lang.Thread.run(Thread.java:1012)
E/MethodChannel#com.bbflight.background_downloader(26605): Failed to handle method call
E/MethodChannel#com.bbflight.background_downloader(26605): kotlinx.serialization.SerializationException: Serializer for class 'Any' is not found.
E/MethodChannel#com.bbflight.background_downloader(26605): Please ensure that class is marked as '@Serializable' and that the serialization compiler plugin is applied.
E/MethodChannel#com.bbflight.background_downloader(26605): 
E/MethodChannel#com.bbflight.background_downloader(26605): 	at kotlinx.serialization.internal.Platform_commonKt.serializerNotRegistered(Platform.common.kt:91)
E/MethodChannel#com.bbflight.background_downloader(26605): 	at kotlinx.serialization.SerializersKt__SerializersKt.noCompiledSerializer(Serializers.kt:366)
E/MethodChannel#com.bbflight.background_downloader(26605): 	at kotlinx.serialization.SerializersKt.noCompiledSerializer(Unknown Source:1)
E/MethodChannel#com.bbflight.background_downloader(26605): 	at com.bbflight.background_downloader.BDPlugin.methodUpdateChunkStatus(BDPlugin.kt:96)
E/MethodChannel#com.bbflight.background_downloader(26605): 	at com.bbflight.background_downloader.BDPlugin.access$methodUpdateChunkStatus(BDPlugin.kt:56)
E/MethodChannel#com.bbflight.background_downloader(26605): 	at com.bbflight.background_downloader.BDPlugin$onMethodCall$1.invokeSuspend(BDPlugin.kt:392)
E/MethodChannel#com.bbflight.background_downloader(26605): 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
E/MethodChannel#com.bbflight.background_downloader(26605): 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
E/MethodChannel#com.bbflight.background_downloader(26605): 	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:280)
E/MethodChannel#com.bbflight.background_downloader(26605): 	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
E/MethodChannel#com.bbflight.background_downloader(26605): 	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
E/MethodChannel#com.bbflight.background_downloader(26605): 	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source:1)
E/MethodChannel#com.bbflight.background_downloader(26605): 	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
E/MethodChannel#com.bbflight.background_downloader(26605): 	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source:1)
E/MethodChannel#com.bbflight.background_downloader(26605): 	at com.bbflight.background_downloader.BDPlugin.onMethodCall(BDPlugin.kt:376)
E/MethodChannel#com.bbflight.background_downloader(26605): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:267)
E/MethodChannel#com.bbflight.background_downloader(26605): 	at io.flutter.embedding.engine.dart.DartMessenger.invokeHandler(DartMessenger.java:292)
E/MethodChannel#com.bbflight.background_downloader(26605): 	at io.flutter.embedding.engine.dart.DartMessenger.lambda$dispatchMessageToQueue$0(DartMessenger.java:319)
E/MethodChannel#com.bbflight.background_downloader(26605): 	at io.flutter.embedding.engine.dart.DartMessenger.$r8$lambda$2j2MERcK825A5j1fv5sZ7xB2Iuo(Unknown Source:0)
E/MethodChannel#com.bbflight.background_downloader(26605): 	at io.flutter.embedding.engine.dart.DartMessenger$$ExternalSyntheticLambda0.run(Unknown Source:12)
E/MethodChannel#com.bbflight.background_downloader(26605): 	at android.os.Handler.handleCallback(Handler.java:958)
E/MethodChannel#com.bbflight.background_downloader(26605): 	at android.os.Handler.dispatchMessage(Handler.java:99)
E/MethodChannel#com.bbflight.background_downloader(26605): 	at android.os.Looper.loopOnce(Looper.java:257)
E/MethodChannel#com.bbflight.background_downloader(26605): 	at android.os.Looper.loop(Looper.java:368)
E/MethodChannel#com.bbflight.background_downloader(26605): 	at android.app.ActivityThread.main(ActivityThread.java:8839)
E/MethodChannel#com.bbflight.background_downloader(26605): 	at java.lang.reflect.Method.invoke(Native Method)
E/MethodChannel#com.bbflight.background_downloader(26605): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:572)
E/MethodChannel#com.bbflight.background_downloader(26605): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1049)
```

other the log maybe relate : 

```
I/flutter (26605): PlatformException(error, Serializer for class 'Any' is not found.
I/flutter (26605): Please ensure that class is marked as '@Serializable' and that the serialization compiler plugin is applied.
I/flutter (26605): , null, kotlinx.serialization.SerializationException: Serializer for class 'Any' is not found.
I/flutter (26605): Please ensure that class is marked as '@Serializable' and that the serialization compiler plugin is applied.
I/flutter (26605): 
I/flutter (26605): 	at kotlinx.serialization.internal.Platform_commonKt.serializerNotRegistered(Platform.common.kt:91)
I/flutter (26605): 	at kotlinx.serialization.SerializersKt__SerializersKt.noCompiledSerializer(Serializers.kt:366)
I/flutter (26605): 	at kotlinx.serialization.SerializersKt.noCompiledSerializer(Unknown Source:1)
I/flutter (26605): 	at com.bbflight.background_downloader.BDPlugin.methodUpdateChunkStatus(BDPlugin.kt:96)
I/flutter (26605): 	at com.bbflight.background_downloader.BDPlugin.access$methodUpdateChunkStatus(BDPlugin.kt:56)
I/flutter (26605): 	at com.bbflight.background_downloader.BDPlugin$onMethodCall$1.invokeSuspend(BDPlugin.kt:392)
I/flutter (26605): 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
I/flutter (26605): 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
I/flutter (26605): 	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:280)
I/flutter (26605): 	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
I/flutter (26605): 	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
I/flutter (26605): 	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source:1)
I/flutter (26605): 	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
I/flutter (26605): 	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source:1)
I/flutter (26605): 	at com.bbflight.background_downloader.BDPlugin.onMethodCall(BDPlugin.kt:376)
I/flutter (26605): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:267)
I/flutter (26605): 	at io.flutter.embedding.engine.dart.DartMessenger.invokeHandler(DartMessenger.java:292)
I/flutter (26605): 	at io.flutter.embedding.engine.dart.DartMessenger.lambda$dispatchMessageToQueue$0(DartMessenger.java:319)
I/flutter (26605): 	at io.flutter.embedding.engine.dart.DartMessenger.$r8$lambda$2j2MERcK825A5j1fv5sZ7xB2Iuo(Unknown Source:0)
I/flutter (26605): 	at io.flutter.embedding.engine.dart.DartMessenger$$ExternalSyntheticLambda0.run(Unknown Source:12)
I/flutter (26605): 	at android.os.Handler.handleCallback(Handler.java:958)
I/flutter (26605): 	at android.os.Handler.dispatchMessage(Handler.java:99)
I/flutter (26605): 	at android.os.Looper.loopOnce(Looper.java:257)
I/flutter (26605): 	at android.os.Looper.loop(Looper.java:368)
I/flutter (26605): 	at android.app.ActivityThread.main(ActivityThread.java:8839)
I/flutter (26605): 	at java.lang.reflect.Method.invoke(Native Method)
I/flutter (26605): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:572)
I/flutter (26605): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1049)
I/flutter (26605): )
I/flutter (26605): error from custom zone : PlatformException(error, Serializer for class 'Any' is not found.
I/flutter (26605): Please ensure that class is marked as '@Serializable' and that the serialization compiler plugin is applied.
I/flutter (26605): , null, kotlinx.serialization.SerializationException: Serializer for class 'Any' is not found.
I/flutter (26605): Please ensure that class is marked as '@Serializable' and that the serialization compiler plugin is applied.
I/flutter (26605): 
I/flutter (26605): 	at kotlinx.serialization.internal.Platform_commonKt.serializerNotRegistered(Platform.common.kt:91)
I/flutter (26605): 	at kotlinx.serialization.SerializersKt__SerializersKt.noCompiledSerializer(Serializers.kt:366)
I/flutter (26605): 	at kotlinx.serialization.SerializersKt.noCompiledSerializer(Unknown Source:1)
I/flutter (26605): 	at com.bbflight.background_downloader.BDPlugin.methodUpdateChunkStatus(BDPlugin.kt:96)
I/flutter (26605): 	at com.bbflight.background_downloader.BDPlugin.access$methodUpdateChunkStatus(BDPlugin.kt:56)
I/flutter (26605): 	at com.bbflight.background_downloader.BDPlugin$onMethodCall$1.invokeSuspend(BDPlugin.kt:392)
I/flutter (26605): 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
I/flutter (26605): 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
I/flutter (26605): 	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:280)
I/flutter (26605): 	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
I/flutter (26605): 	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
I/flutter (26605): 	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source:1)
I/flutter (26605): 	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
I/flutter (26605): 	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source:1)
I/flutter (26605): 	at com.bbflight.background_downloader.BDPlugin.onMethodCall(BDPlugin.kt:376)
I/flutter (26605): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:267)
I/flutter (26605): 	at io.flutter.embedding.engine.dart.DartMessenger.invokeHandler(DartMessenger.java:292)
I/flutter (26605): 	at io.flutter.embedding.engine.dart.DartMessenger.lambda$dispatchMessageToQueue$0(DartMessenger.java:319)
I/flutter (26605): 	at io.flutter.embedding.engine.dart.DartMessenger.$r8$lambda$2j2MERcK825A5j1fv5sZ7xB2Iuo(Unknown Source:0)
I/flutter (26605): 	at io.flutter.embedding.engine.dart.DartMessenger$$ExternalSyntheticLambda0.run(Unknown Source:12)
I/flutter (26605): 	at android.os.Handler.handleCallback(Handler.java:958)
I/flutter (26605): 	at android.os.Handler.dispatchMessage(Handler.java:99)
I/flutter (26605): 	at android.os.Looper.loopOnce(Looper.java:257)
I/flutter (26605): 	at android.os.Looper.loop(Looper.java:368)
I/flutter (26605): 	at android.app.ActivityThread.main(ActivityThread.java:8839)
I/flutter (26605): 	at java.lang.reflect.Method.invoke(Native Method)
I/flutter (26605): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:572)
I/flutter (26605): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1049)
I/flutter (26605): )
```

and there is 2 task has successful:

```
I/System.out(26605): [OkHttp] sendRequest>>
I/System.out(26605): [OkHttp] sendRequest<<
I/TaskWorker(26605): Successfully downloaded taskId 28401318 to /data/user/0/******/app_flutter/com.bbflight.background_downloader.1250690910
I/WM-WorkerWrapper(26605): Worker result SUCCESS for Work [ id=2ca6d5c6-0042-4e4c-bd0d-c6c1cb74223b, tags={ com.bbflight.background_downloader.DownloadTaskWorker, BackgroundDownloader, taskId=28401318, group=chunk } ]
I/TaskWorker(26605): Successfully downloaded taskId 1652368664 to /data/user/0/******/app_flutter/com.bbflight.background_downloader.1094219686
I/WM-WorkerWrapper(26605): Worker result SUCCESS for Work [ id=cbcd1e84-8461-478e-b674-dcbe795910bb, tags={ com.bbflight.background_downloader.DownloadTaskWorker, BackgroundDownloader, taskId=1652368664, group=chunk } ]
```

I think maybe is one of theme trigger an exception, and failed full download task ?
in this situation, I cant get any thing from 'Future<TaskStatusUpdate>().then()'.

**Comment by 781flyingdutchman** [comment]
Thanks - what plugin version are you using? Asking because the line referenced in the log makes no sense: 
```
com.bbflight.background_downloader.BDPlugin.methodUpdateChunkStatus(BDPlugin.kt:96)
```

If you look in the repo, that line isn't anywhere near the methodUpdateChunkStatus so I don't quite understand how that can be.

**Comment by bladeofgod** [comment]
> Thanks - what plugin version are you using? Asking because the line referenced in the log makes no sense:
> 
> ```
> com.bbflight.background_downloader.BDPlugin.methodUpdateChunkStatus(BDPlugin.kt:96)
> ```
> 
> If you look in the repo, that line isn't anywhere near the methodUpdateChunkStatus so I don't quite understand how that can be.

v8.6.0 from pubspec.lock file.

**Comment by 781flyingdutchman** [comment]
Fixed in 8.7.0
